server/src/Dao/RevisionDao.php

```diff
@@ -110,4 +112,38 @@ public function getRevisionDetails( $revIds ) {
 
 		return $this->fetchAll( $query, \PDO::FETCH_ASSOC );
 	}
+
+	/**
+	 * Determine user's search column on revisions table.
+	 *
+	 * When possible, search for user ids to improve performance.
+	 * If usernames or IP(s) are provided, then use those.
+	 *
+	 * @param array $users List of usernames/IPs or user ids
+	 * @return string
+	 */
+	private function getUserSearchColumn( $users ) {
+		$column = 'rev_user_text';
+		if ( $this->isNumericArray( $users ) ) {
+			$column = 'rev_user';
+		}
+
+		return $column;
+	}
+
+	/**
+	 * Determine if all the elements in an array are numeric.
+	 *
+	 * @param array $arr
+	 * @return boolean
+	 */
+	private function isNumericArray( $arr ) {
```

Comment:
you have a _similar_ implementation in `isIpPresent` but uses a completely different methodology (loop vs filter). I prefer the filter, but either way, I think it should be consistent (especially in the same PR, ha).

Also, add an `array` typehint.

Reply:
I don't think you need to loop through all of the elements in `isIpPresent` either? Shouldn't it just return as soon as the first IP is found?